### PR TITLE
refactor: create/edit squad form and its types

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes
- Refactored both process to make them behave the same rather than one doing a pre-calculation, and then the other happening outside the mutation function call side.
- While refactoring, I realized some properties are not used anymore and, hence, removed (like commentary).
  - At first, I planned to introduce a separate client-side function to handle squad creation with commentary, but the mutation itself does not have the said property anymore.
- Unified the types as the original one was combining two things at once. The form data and initial squad data on edit.
  - To avoid that, we want to make it clear that anything initial data related, should come from `squad` property. Other than that, it is the form to decide.
  - Due to the previous setup, TypeScript was going crazy due to certain type usages of the Form.

## Events

Did you introduce any new tracking events?

## Experiment

Did you introduce any new experiments?

## Manual Testing

> [!CAUTION]
> Please make sure existing components are not breaking/affected by this PR

### Preview domain
https://refactor-forms.preview.app.daily.dev

### CODE REVIEW

**Comment by sshanzel** [comment]
Root cause on what triggered me to make this PR 😂 https://github.com/dailydotdev/apps/pull/3496/commits/78977915ff645ae6b9106fc9c08983318ecc2a89

File: packages/shared/src/components/squads/Details.tsx [hunk 1]
```diff
@@ -49,25 +48,27 @@ const getFormData = async (
 
   const input = document.getElementById(squadImageId) as HTMLInputElement;
   const file = input.files[0];
-  const base64 = await blobToBase64(file);
 
-  return { ...current, file: base64 };
```
Comment (rebelchris) [inline packages/shared/src/components/squads/Details.tsx:54]: Why is this removed?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Just one query, where did this base64 stuff go now?

File: packages/shared/src/components/squads/Details.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/shared/src/components/squads/Details.tsx:54]: The type requirement of the mutation is a `File`, here we convert it to `string`, then before sending it, we then again turn it into a `File.

Comment (rebelchris) [reply packages/shared/src/components/squads/Details.tsx:54]: Hmm just make sure it doesn't break implementation (as locally the upload is not e2e testable)
And maybe considering this: https://dailydotdev.slack.com/archives/C01L0QXQJNB/p1724993128390989

**Review by rebelchris — APPROVED** [review]
Good to go, but just verify in prod it works with upload specific :)

**Comment by sshanzel** [comment]
> Good to go, but just verify in prod it works with upload specific :)

Yeah, just waiting for the go signal if we can merge without updating the feed yet. After that, I will run some tests.
